src/utils/cardUtilityFunctions.js

```diff
@@ -122,3 +122,171 @@ export const getUpdatedCardSize = oldSize => {
   }
   return newSize;
 };
+
+// re-usable variable fetcher that returns an array of variables
+// variables are identified by surrounding curly braces i.e. {deviceid}
+export const getVariables = value => {
+  let variables = value && typeof value === 'string' ? value.match(/{[a-zA-Z0-9_-]+}/g) : null;
+  if (variables) {
+    variables = variables.map(variable => variable.replace(/[{}]/g, ''));
+  }
+  return variables;
+};
+
+// replace variables from the list of variables that are found on the target with their corresponding value
+export const replaceVariables = (variables, values, target) => {
+  let updatedTarget = target;
+  variables.forEach(variable => {
+    const variableRegex = new RegExp(`{${variable}}`, 'g');
+    updatedTarget = updatedTarget.replace(variableRegex, values[variable]);
+  });
+  return updatedTarget;
+};
+
+export const handleVariables = card => {
+  if (!card.cardVariables) {
+    return card;
+  }
+  const updatedCard = card;
```

Comment:
Need to clone the card first to remove the value by reference

Reply:
Good catch. I will update.